### PR TITLE
#476: Split off native code into different OSGi fragments

### PR DESCRIPTION
**Pull request by ctron**
This change fixes issue #476 by splitting off different native code libraries into fragments which can be selectively installed on the target platform when necessary.

### CODE REVIEW

**Comment by cdealti** [comment]
@ctron This is good. Should we wait until the split is done for every bundle including native libs? In other words, do you plan to update the PR pushing more commits?

**Comment by ctron** [comment]
Well that was my plan. But I have to do the rest anyway. This one is complete. We can merge this and follow up with the others in a separate PR.

From an Equinox 3.8 perspective it doesn't matter anyway.

File: kura/org.eclipse.kura.linux.usb.x86_64/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -3,6 +3,6 @@ Bundle-ManifestVersion: 2
 Bundle-Name: Native libraries for 'org.eclipse.kura.linux.usb' on x86_64
 Bundle-SymbolicName: org.eclipse.kura.linux.usb.x86_64
 Bundle-Version: 1.0.0.qualifier
-Fragment-Host: org.eclipse.kura.linux.usb;bundle-version="1.0.8"
+Fragment-Host: org.eclipse.kura.linux.usb;bundle-version="[1.0.8,2.0.0)"
```
Comment (cdealti) [inline kura/org.eclipse.kura.linux.usb.x86_64/META-INF/MANIFEST.MF]: The above changes should logically go with the previous commit for org.eclipse.kura.linux.usb.

File: target-platform/config/kura.target-platform.build.properties [hunk 2]
```diff
@@ -24,7 +24,7 @@ com.google.protobuf.version=2.6.0
 org.tigris.mtoolkit.iagent.rpc.version=3.0.0.20110411-0918
 com.gwt.user.version=1.0.0
 org.hamcrest.core.version=1.1.0.v20090501071000
-jdk.dio.version=1.0.5
+jdk.dio.version=1.0.6-SNAPSHOT
```
Comment (cdealti) [inline target-platform/config/kura.target-platform.build.properties]: So we only track the version of the parent module jdk.dio in kura.target-platform.build.properties and we don't individually version the fragments, correct?

**Comment by ctron** [comment]
Rebased.

File: target-platform/config/kura.target-platform.build.properties (hunk 2 above)
Comment (ctron) [inline target-platform/config/kura.target-platform.build.properties]: Yes.

**Comment by cdealti** [comment]
@mmaiero please review and try to understand the changes needed for our ESF dependent build.

File: kura/distrib/config/kura.build.properties [hunk 3]
```diff
@@ -34,6 +34,12 @@ org.eclipse.kura.linux.net.version=1.0.11-SNAPSHOT
 org.eclipse.kura.linux.net.test.version=1.0.4-SNAPSHOT
 org.eclipse.kura.linux.position.version=1.0.7-SNAPSHOT
 org.eclipse.kura.linux.usb.version=1.0.8-SNAPSHOT
+org.eclipse.kura.linux.usb.armv5sf.version=1.0.0-SNAPSHOT
```
Comment (MMaiero) [inline kura/distrib/config/kura.build.properties]: Hi Jens,
why the fragments, in this case, have a version that is different of the  host one? This is different if compared to jdk.dio.

File: kura/distrib/pom.xml [hunk 4]
```diff
@@ -393,6 +424,12 @@
...
+							    <move file="target/plugins/org.eclipse.kura.linux.usb.armv5sf.jar" tofile="target/plugins/org.eclipse.kura.linux.usb.armv5sf_${org.eclipse.kura.linux.usb.armv5sf.version}.jar"/>
+							    <move file="target/plugins/org.eclipse.kura.linux.usb.armv6hf.jar" tofile="target/plugins/org.eclipse.kura.linux.usb.armv6hf_${org.eclipse.kura.linux.usb.armv6hf.version}.jar"/>
+							    <move file="target/plugins/org.eclipse.kura.linux.usb.x86.jar" tofile="target/plugins/org.eclipse.kura.linux.usb.x86_${org.eclipse.kura.linux.usb.x86.version}.jar"/>
+							    <move file="target/plugins/org.eclipse.kura.linux.usb.x86_64.jar" tofile="target/plugins/org.eclipse.kura.linux.usb.x86_64_${org.eclipse.kura.linux.usb.x86_64.version}.jar"/>
+							    <move file="target/plugins/org.eclipse.kura.linux.usb.x86_64.yocto.jar" tofile="target/plugins/org.eclipse.kura.linux.usb.x86_64.yocto_${org.eclipse.kura.linux.usb.x86_64.yocto.version}.jar"/>
+							    <move file="target/plugins/org.eclipse.kura.linux.usb.x86.ubuntu.jar" tofile="target/plugins/org.eclipse.kura.linux.usb_${org.eclipse.kura.linux.usb.x86.ubuntu.version}.jar"/>
```
Comment (MMaiero) [inline kura/distrib/pom.xml]: The toFile here is wrong

File: kura/distrib/src/main/ant/build_equinox_distrib.xml [hunk 5]
```diff
@@ -301,6 +302,10 @@ fi]]>
 		<replaceregexp file="${project.build.directory}/${build.output.name}/kura.init.raspbian" match="INSTALL_DIR=.*" replace="INSTALL_DIR=${kura.install.dir}" />
 		<replaceregexp file="${project.build.directory}/${build.output.name}/kura.service.edison" match="INSTALL_DIR" replace="${kura.install.dir}" />
 
+		<script language="javascript">
+	    var nativeTag = project.getProperty("native.tag");
+	    project.setNewProperty("org.eclipse.linux.usb.native.version", project.getProperty("org.eclipse.linux.usb." + nativeTag + ".version" ) );
```
Comment (MMaiero) [inline kura/distrib/src/main/ant/build_equinox_distrib.xml]: I tried to build the code, but Kura didn't started. I've found that this setNewProperty was not correctly working and the config.ini file was trying to start a bundle with version ${org.eclipse.linux.usb.native.version}

File: kura/distrib/config/kura.build.properties (hunk 3 above)
Comment (ctron) [inline kura/distrib/config/kura.build.properties]: I am fine collapsing them into one.

File: kura/distrib/pom.xml (hunk 4 above)
Comment (ctron) [inline kura/distrib/pom.xml]: I see.

File: kura/distrib/src/main/ant/build_equinox_distrib.xml (hunk 5 above)
Comment (ctron) [inline kura/distrib/src/main/ant/build_equinox_distrib.xml]: That code path is gone with the alignment of the versions.

**Comment by ctron** [comment]
I did amend the PR addressing the comments of @MMaiero

**Comment by ctron** [comment]
@MMaiero We can close this one if it was in your bigger merge, right?

**Pull request closed without merging**